apps/emqx_connector/test/emqx_connector_redis_SUITE.erl

```diff
@@ -0,0 +1,135 @@
+% %%--------------------------------------------------------------------
+% %% Copyright (c) 2020-2022 EMQ Technologies Co., Ltd. All Rights Reserved.
+% %%
+% %% Licensed under the Apache License, Version 2.0 (the "License");
+% %% you may not use this file except in compliance with the License.
+% %% You may obtain a copy of the License at
+% %% http://www.apache.org/licenses/LICENSE-2.0
+% %%
+% %% Unless required by applicable law or agreed to in writing, software
+% %% distributed under the License is distributed on an "AS IS" BASIS,
+% %% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% %% See the License for the specific language governing permissions and
+% %% limitations under the License.
+% %%--------------------------------------------------------------------
+
+-module(emqx_connector_redis_SUITE).
+
+-compile(nowarn_export_all).
+-compile(export_all).
+
+-include_lib("eunit/include/eunit.hrl").
+-include_lib("emqx/include/emqx.hrl").
+-include_lib("stdlib/include/assert.hrl").
+
+-define(REDIS_HOST, "redis").
+-define(REDIS_PORT, 6379).
+
+all() ->
+    emqx_common_test_helpers:all(?MODULE).
+
+groups() ->
+    [].
+
+init_per_suite(Config) ->
+    case emqx_common_test_helpers:is_tcp_server_available(?REDIS_HOST, ?REDIS_PORT) of
+        true ->
+            ok = emqx_connector_test_helpers:start_apps([ecpool, eredis]),
+            Config;
+        false ->
+            {skip, no_redis}
+    end.
+
+end_per_suite(_Config) ->
+    ok = emqx_connector_test_helpers:stop_apps([ecpool, eredis]),
+    ok.
+
+init_per_testcase(_, Config) ->
+    ?assertEqual(
+        {ok, #{poolname => emqx_connector_redis, type => single}},
+        emqx_connector_redis:on_start(<<"emqx_connector_redis">>, redis_config())
+    ),
+    Config.
+
+end_per_testcase(_, _Config) ->
+    ?assertEqual(
+        ok,
+        emqx_connector_redis:on_stop(<<"emqx_connector_redis">>, #{poolname => emqx_connector_redis})
+    ).
+
+% %%------------------------------------------------------------------------------
+% %% Testcases
+% %%------------------------------------------------------------------------------
+
+% Simple test to make sure the proper reference to the module is returned.
+t_roots(_Config) ->
+    ExpectedRoots = [
+        {config, #{
+            type =>
+                {union, [
+                    {ref, emqx_connector_redis, cluster},
+                    {ref, emqx_connector_redis, single},
+                    {ref, emqx_connector_redis, sentinel}
+                ]}
+        }}
+    ],
+    ActualRoots = emqx_connector_redis:roots(),
+    ?assertEqual(ExpectedRoots, ActualRoots).
```

Comment:
We'd better test the schema definition by providing different kinds of configs to `emqx_connector_redis:on_start/2`.

The hocon schema for the connector module is probably be changed in the future, but the way we calling `emqx_connector_redis:on_start/2` and `emqx_connector_redis:on_start/3` won't.

Reply:
So would you prefer this test be removed and the on_start to be called multiple times with different configs even if it doesn't change much/anything about our actual code?